Model/Events/CatalogSearch.php

```diff
@@ -5,13 +5,21 @@
 class CatalogSearch
 {
     public function __construct(
-        \Magento\Search\Helper\Data     $searchHelper,
-        \Magento\Framework\UrlInterface $urlInterface
+        \Magento\Framework\UrlInterface     $urlInterface,
+        \Magento\Framework\App\Request\Http $request
     ) {
-        $this->searchHelper = $searchHelper;
         $this->urlInterface = $urlInterface;
+        $this->request      = $request;
     }
     public function callJS() {
-        return "window.metrilo.search('" . $this->searchHelper->getEscapedQueryText() . "', '" . $this->urlInterface->getCurrentUrl() . "');";
+        $searchQuery = $this->request->getParam('q');
```

Comment:
Would this work on a Magento 2 installation with different language?
What was wrong with the previous `searchHelper->getEscapedQueryText` approach?

Reply:
The problem comes from the native search module of the magento system that both search and advanced search are using. In the search module helper is used a hardcoded variable (https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Search/Helper/Data.php#L231 and the actual value is in another file https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Search/Model/QueryFactory.php#L23) that allows only 'q' param to be extracted from the request, thus limiting it to normal search usage only since there is no 'q' param being attached to the request when using advanced search (name, sku, description, short_description, price[from], price[to] are the params attached from the advanced search).

Reply:
> Would this work on a Magento 2 installation with different language?

Reply:
Yes, if the core system files are not modified it should work on any language setup.